docs/building-a-form/about-the-json-schema-standard-and-react-jsonschema-form.md

```diff
@@ -0,0 +1,298 @@
+
+The schemaform uses these custom fields and widgets:
+
+- `YesNoWidget`
+- `AddressField`
+- `DateWidget`
+- `SSNWidget`
+- `PhoneWidget`
+
+Writing custom widgets is similar to writing React components: A value is passed in, and an `onChange` hook is provided for changing data. Other properties like the schemas and field ID are also provided.
+
+Custom fields receive all properties listed previously for field components in rjsf.
+
+In addition to customizing fields and widgets, the schemaform code hooks into a number of events provided by `Form` to support our form patterns, found in the `FormPage` component. These events are:
+
+- `validate`: This event is called when validation occurs. We call our custom validation, which reads uiSchema for custom validation hooks that have been included for form fields outside of what JSON Schema provides.
+- `transformErrors`: This event is provided when the schemaform receives the list of JSON Schema validation errors and can return a transformed list. It replaces the messages with a set of default messages, as well as any messages provided for specific fields in uiSchema. It also moves the errors for required fields from the object level to the field level. Because JSON Schema specifies required fields with a `required` array on an object field schema, any errors about missing data are associated with that object and moved so they're associated with the missing field and rendered with that field on the form.
+- `onError`: This event is called if a user tries to submit a form with a validation error. The schemaform sets a `submitted` flag in `formContext`, which is an object passed to all fields and components in the rjsf form. The `FieldTemplate` component uses `formContext` to display all error messages to the user.
+- `onSubmit`: This event is called when a user submits a form with no validation errors. When this happens, the schemaform code looks for the next page in the multi-page form and navigates to it.
+- `onChange`: This event is called when a user changes data in the form. The schemaform fires a Redux action and updates the store with the new data. The reducer code does several recalculations:
+    - **Recalculate the required fields for the schema:** You can specify functions in uiSchema that set fields as optional or required based on form data. This runs them and updates the schema.
+    - **Recalculate which schema fields are hidden and remove that data:** In uiSchema, you can specify fields that are conditionally hidden based on user data. To avoid validation errors from data a user can't see, the schemaform updates the schema to add a `ui:hidden` property and remove any user data for those fields.
+    - **Recalcuate general schema updates:** Because you can make arbitrary changes to the schema based on form data, the schemaform must also make those changes, for example, removing options in an `enum` array when a user has entered certain data.
+
+##### Supporting multi-page forms
```

Comment:
I'd originally proposed putting this in a separate topic, but realized this information was simply explaining how rjsf handles the need for multi-page forms, rather than the form builder needing to build multi-page forms. However, I do think we may want a topic with guidance for building multi-page forms.